test/test_bucket.py

```diff
@@ -276,6 +277,95 @@ def test_upload_large(self):
         self._check_file_contents('file1', data)
         self.assertEqual("600: 200 400 600", progress_listener.get_history())
 
+    def test_upload_large_resume(self):
+        part_size = self.simulator.MIN_PART_SIZE
+        data = self._make_data(part_size * 3)
+        large_file_id = self._start_large_file('file1')
+        self._upload_part(large_file_id, 1, data[:part_size])
+        progress_listener = StubProgressListener()
+        file_info = self.bucket.upload_bytes(data, 'file1', progress_listener=progress_listener)
+        self.assertEqual(large_file_id, file_info.id_)
+        self._check_file_contents('file1', data)
+        self.assertEqual("600: 200 400 600", progress_listener.get_history())
+
+    def test_upload_large_resume_no_parts(self):
+        part_size = self.simulator.MIN_PART_SIZE
+        data = self._make_data(part_size * 3)
+        large_file_id = self._start_large_file('file1')
+        progress_listener = StubProgressListener()
+        file_info = self.bucket.upload_bytes(data, 'file1', progress_listener=progress_listener)
+        self.assertNotEqual(large_file_id, file_info.id_)  # it's not a match if there are no parts
+
+    def test_upload_large_resume_all_parts_there(self):
+        part_size = self.simulator.MIN_PART_SIZE
+        data = self._make_data(part_size * 3)
+        large_file_id = self._start_large_file('file1')
+        self._upload_part(large_file_id, 1, data[:part_size])
+        self._upload_part(large_file_id, 2, data[part_size:2 * part_size])
+        self._upload_part(large_file_id, 3, data[2 * part_size:])
+        progress_listener = StubProgressListener()
+        file_info = self.bucket.upload_bytes(data, 'file1', progress_listener=progress_listener)
+        self.assertEqual(large_file_id, file_info.id_)
+        self._check_file_contents('file1', data)
+        self.assertEqual("600: 200 400 600", progress_listener.get_history())
+
+    def test_upload_large_resume_part_does_not_match(self):
+        part_size = self.simulator.MIN_PART_SIZE
+        data = self._make_data(part_size * 3)
+        large_file_id = self._start_large_file('file1')
+        self._upload_part(large_file_id, 3, data[:part_size])  # wrong part number for this data
+        progress_listener = StubProgressListener()
+        file_info = self.bucket.upload_bytes(data, 'file1', progress_listener=progress_listener)
+        self.assertNotEqual(large_file_id, file_info.id_)
+
```

Comment:
please add `_check_file_contents` to check if the data which should have been uploaded, has been actually uploaded and has valid contents